pkg/upgrade/control_plane.go

```diff
@@ -681,6 +689,24 @@ func (u *ControlPlaneUpgrader) updateAndUploadKubeadmKubernetesVersion() error {
 	return nil
 }
 
+func (u *ControlPlaneUpgrader) removeNodeFromKubeadmConfigMap(nodeName string) error {
+	original, err := u.targetKubernetesClient.CoreV1().ConfigMaps("kube-system").Get("kubeadm-config", metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error getting kubeadm configmap from target cluster")
+	}
+
+	updated, err := removeNodeFromClusterStatus(original, nodeName)
+	if err != nil {
+		return err
+	}
+
+	if _, err = u.targetKubernetesClient.CoreV1().ConfigMaps("kube-system").Update(updated); err != nil {
```

Comment:
Can we Patch instead of Update? That might help with the v1beta1 vs v1beta2 issue?

Reply:
Example of patch: https://github.com/vmware/cluster-api-upgrade-tool/blob/master/pkg/upgrade/machine_deployment.go#L139

Though, there's a github issue to swap the raw controller runtime patches with cluster-api patch helper: https://github.com/vmware/cluster-api-upgrade-tool/issues/83

I'm cool with either.

Reply:
I think we'd have to patch if we wanted v1beta1, because the configmap itself has v1beta2 in the ClusterStatus field.

I'll try a patch. Time to remember how patching works with clientsets! :D

Reply:
Oh it's encoded yaml inside the configmap. Hmm. Patch won't help w/that. But unstructured.Unstructured will.

Reply:
Or create a new go struct that has the json tags you need on it and use that instead of the one in kubeadm.

Reply:
we're already using Update in the codebase for other kubeadm config map manipulation: https://github.com/vmware/cluster-api-upgrade-tool/blob/master/pkg/upgrade/control_plane.go#L684

maybe best to keep them the same for now?

Reply:
Yes, Update is fine to keep. Sorry for the noise.